src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ClientService.java

```diff
@@ -418,6 +416,23 @@ public ClientType deleteTlsCertificate(ClientId id, String certificateHash)
         return mergeClientListsDistinctively(globalClients, localClients);
     }
 
+    /**
+     * Registers a client
+     * @param clientId client to register
+     * @throws GlobalConfOutdatedException
+     * @throws ClientNotFoundException
+     */
+    public void registerClient(ClientId clientId) throws GlobalConfOutdatedException, ClientNotFoundException {
+        ClientType client = getLocalClientOrThrowNotFound(clientId); // throws if client doesn't exist
+        try {
+            managementRequestSenderService.sendClientRegisterRequest(clientId);
+            client.setClientStatus(ClientType.STATUS_REGINPROG);
```

Comment:
This should probably verify that client can be registered, and ultimately result in http 409 if not?

It looks like client can be registered only if status = SAVED. Old implementation either

- registers right after client has been added, and hence has status SAVED https://github.com/nordic-institute/X-Road/blob/19578462600ce3ec1832e3af5ff2b0f1b1cfe01c/src/proxy-ui/public/javascripts/clients.js#L146-L187 or
- shows "register" button only if status is SAVED https://github.com/nordic-institute/X-Road/blob/19578462600ce3ec1832e3af5ff2b0f1b1cfe01c/src/proxy-ui/app/controllers/clients_controller.rb#L476-L477

There is no check in Ruby controllers, but adding this check to service layer would be consistent with how e.g. token register is now implemented.

Reply:
Added a check for this